### DISCUSSION TITLE
Usb midi controller recognized but does not function

### DISCUSSION THREAD

**Discussion by jlam9801** [probonopd/MiniDexed#711] [General]
My Hardware: pi zero 2 w, i2s dac, lcd1602 (i2c), yamaha ypt400 (controller, 61 key, usb midi, functions both on my pc and phone perfectly fine)

As title states, my pi boots to minidexed, recognizes my midi controller but fails to initialize it. Can anyone please help? Will post a picture of the HDMI console output momentarily.

**Comment by jlam9801** [comment]
Here is an image of the HDMI console output 

https://imgur.com/a/PtPulL0


Edited to add: here's a Pic of my build in a temporary case

https://imgur.com/a/lm3GGdj

**Comment by Banana71** [comment]
Hello jlam9801,

your controller keyboard must be Class Compliant. https://github.com/probonopd/MiniDexed/wiki/Hardware#usb-midi-devices

_If your device - such as your MIDI controller or audio interface - is USB Class Compliant, it means you can plug it into your computer and it will work without the need for any other special software_

The manual for your Yamaha PYT400 (PSR-403) states that a driver must be installed in order to use the USB interface. :-( 



The following could help with a Class Compliant problem:
- Another USB keyboard (Class Compliant)
- A USB to MIDI interface
- You build yourself a classic MIDI input for your miniDexed https://github.com/probonopd/MiniDexed/wiki/Hardware#classic-midi-input


Peter

**Reply by jlam9801** [reply]
I hoped that wasn't the case but if so I'll have to get another controller. Using a pc keyboard right now which isn't perfect but works. 

I assumed that the ypt400 was class compliant because I can use it with my galaxy s21 as a midi controller with no driver related issues. Plugged it in with an otg adapter and it just worked. I do remember installing a driver and some related software on my windows machine, but thought that was just related to transferring midi files to the keyboard for playback.

I am going to try the portatone with raspbian as well to see if that works, will report back once I do.
